### PR TITLE
Changing outdated HTTP to HTTPS calls within the codebase

### LINKED ISSUE #4072: tappedout.net deck analysis not working
<b>System Information:</b>
Client Version: 2.7.4 (2020-03-20)
Client Operating System: Windows 10 (10.0)
Build Architecture: 64-bit
Qt Version: 5.12.6
System Locale: en_US
Install Mode: Standard
_______________________________________________________________________________________

Sending a deck to tappedout.net for analysis results in an invalid URL in the browser:
`http://www.tappedout.nethttps//tappedout.net/mtg-decks/paste/`
Tested with Firefox and Edge, and multiple decks.

Relevant debug log entry:
`Tappedout: good reply, http status 301 location "https://tappedout.net/mtg-decks/paste/"`

Perhaps an issue with the redirection from http to https?

### PR DESCRIPTION
**Pull request by ZeldaZach**
## Related Ticket(s)
- Fixes #4072

## Short roundup of the initial problem
Bad redirect support within the code

## What will change with this Pull Request?
Change HTTP -> HTTPS for relevant operations. Could add better checks in the future, but trying to fix bugs before release

### CODE REVIEW

File: common/decklist.cpp [hunk 1]
```diff
@@ -336,7 +336,7 @@ DeckList::DeckList()
     root = new InnerDecklistNode;
 }
 
-// TODO: http://qt-project.org/doc/qt-4.8/qobject.html#no-copy-constructor-or-assignment-operator
+// TODO: https://qt-project.org/doc/qt-4.8/qobject.html#no-copy-constructor-or-assignment-operator
```
Comment (ebbit1q) [inline common/decklist.cpp:339]: this is a dead link

Comment (ebbit1q) [reply common/decklist.cpp:339]: https://doc.qt.io/qt-5/qobject.html#no-copy-constructor-or-assignment-operator if you wanted to update this comment, I'm not sure what the problem is though.

Comment (ZeldaZach) [reply common/decklist.cpp:339]: It'll auto redirect

Comment (ebbit1q) [reply common/decklist.cpp:339]: no, because the qt-project.org domain uses an old cert it seems

**Pull request merged**